test/client.test.js

```diff
@@ -0,0 +1,292 @@
+describe('Sharded Client tests', function () {
+  var async = require('async');
+  var MockRedisClient = require('./mockRedisClient.js');
+  var proxyquire = require('proxyquire');
+
+  var shardable = [
+    'sadd',
+    'expire',
+    'ttl',
+    'sismember',
+    'srem',
+    'get',
+    'mget',
+    'exists',
+    'scard',
+    'smembers',
+    'sunion', // sunion assumes that each key is on the same shard
+    'hdel',
+    'hget',
+    'hincrby',
+    'hset',
+    'hmset',
+    'hgetall',
+    'llen',
+    'lpush',
+    'lrange',
+    'ltrim',
+    'set',
+    'setnx',
+    'setex',
+    'psetex',
+    'del',
+    'srandmember',
+    'zrevrange',
+    'incr',
+    'zadd',
+    'zcard',
+    'zcount',
+    'zrem',
+    'zscore',
+    'zrange',
+    'zrangebyscore',
+    'zremrangebyrank'
+  ];
+
+  var readOnly = [
+    'sismember',
+    'get',
+    'mget',
+    'exists',
+    'hgetall',
+    'llen',
+    'lrange',
+    'scard',
+    'smembers',
+    'srandmember',
+    'sunion', // sunion assumes that each key is on the same shard
+    'zrevrange',
+    'zcard',
+    'zcount',
+    'zscore',
+    'zrange',
+    'zrangebyscore'
+  ];
+
+  // Reset before each test
+  var ShardedRedis;
+
+  /********************************
+   * Setup
+   ********************************/
+
+  beforeEach(function () {
+    // Get sharded-redis-client with the redis clients stubbed out
+    ShardedRedis = proxyquire('../sharded-redis-client.js', {
+      redis: MockRedisClient
+    });
+  });
+
+  /********************************
+   * Tests
+   ********************************/
+
+  it('should create a node_redis client when created', function () {
+    spyOn(MockRedisClient, 'createClient').and.callThrough();
+    var numMasterHosts = 1;
+    var options = { redisOptions: {} };
+    var redisHosts = generateRedisHosts(numMasterHosts);
+
+    var shardedClient = new ShardedRedis(redisHosts, options);
+    expect(MockRedisClient.createClient).toHaveBeenCalledTimes(1);
+    expect(MockRedisClient.createClient).toHaveBeenCalledWith(redisHosts[0].port_range[0], redisHosts[0].host, options.redisOptions);
```

Comment:
This also seems a _little_ too bound to the implementation of the sharded redis client. It would be ideal if we could test instead for some form of output, rather than the args that were passed to a specific method.

E.g. I know this probably doesn't work, but:

`expect(shardedClient.port_range).toEqual(redisHosts[0].port_range[0]);`